src/mem_table.rs

```diff
@@ -183,6 +182,12 @@ impl<T: RangeBounds<KVTableInternalKey>> KeyValueIterator for MemTableIterator<'
     async fn next_entry(&mut self) -> Result<Option<RowEntry>, SlateDBError> {
         Ok(self.next_entry_sync())
     }
+
+    async fn seek(&mut self, next_key: &[u8]) -> Result<(), SlateDBError> {
+        Err(SlateDBError::Unsupported(
+            "seek is not supported for memtable iterator".to_string(),
+        ))
```

Comment:
this seek seems not being called in the current code path.

this unsupported can be removed after #543 merged